mod/search/views/default/resources/search/index.php

```diff
@@ -61,6 +61,27 @@
 	$order = 'desc';
 }
 
+// Get object containers: container of comments
+$container_guids = array();
+if (!empty($container_guid)) {
+	$container_guids[] = (int)$container_guid; // Add group container to the container's list
+	$comment_subtype_id = get_subtype_id('object', 'comment');
+	if (!empty($comment_subtype_id)) {
+		$params = array(
+			'type' => 'object',
+			'container_guid' => (int)$container_guid,
+			'joins' => ["LEFT JOIN elgg_entities AS ce ON ce.container_guid = e.guid AND ce.type = 'object' AND ce.subtype = $comment_subtype_id"],
+			'wheres' => ["ce.guid IS NOT NULL"],
+			'limit' => 0,
+		);
+		$object_containers = elgg_get_entities($params);
```

Comment:
this potentially has a lot of entities returned. This could cause Out Of Memory (OOM) issues
Eighter use an ElggBatch (with `$options['batch'] => true`) or use a custom callback to only return the row guid
```
$options['callback'] = function($row){
   return (int) $row->guid;
}
```
and simply merge `$container_guids` and `$object_containers`

Reply:
Thanks for the advice, I will fix it.

Reply:
Shouldn't we implement both improvements: batch and callback ?

Reply:
I did both

Reply:
Bad idea: usage is different, I remove batch.